From c3cae52499ecc3925effc78106eab3da0e824f0e Mon Sep 17 00:00:00 2001
From: Zac <zac@zacbowling.com>
Date: Wed, 14 Jan 2026 19:50:45 -0800
Subject: [PATCH 02/23] wifi: mt76: mt7925: fix missing mutex protection in
 reset and ROC abort

Add missing mt792x_mutex_acquire/release calls around
ieee80211_iterate_active_interfaces() in mt7925_mac_reset_work()
to prevent potential race conditions during chip reset recovery.

Signed-off-by: Zac Bowling <zac@zacbowling.com>
---
 drivers/net/wireless/mediatek/mt76/mt7925/mac.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/mediatek/mt76/mt7925/mac.c b/drivers/net/wireless/mediatek/mt76/mt7925/mac.c
index 958f575647..8c0d375a06 100644
--- a/drivers/net/wireless/mediatek/mt76/mt7925/mac.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7925/mac.c
@@ -1330,12 +1330,12 @@ void mt7925_mac_reset_work(struct work_struct *work)
 	dev->hw_full_reset = false;
 	pm->suspended = false;
 	ieee80211_wake_queues(hw);
+
+	mt792x_mutex_acquire(dev);
 	ieee80211_iterate_active_interfaces(hw,
 					    IEEE80211_IFACE_ITER_RESUME_ALL,
 					    mt7925_vif_connect_iter, NULL);
 	mt76_connac_power_save_sched(&dev->mt76.phy, pm);
-
-	mt792x_mutex_acquire(dev);
 	mt7925_mcu_set_clc(dev, "00", ENVIRON_INDOOR);
 	mt792x_mutex_release(dev);
 }
-- 
2.52.0

